EngineLayer/PsmTsv/PsmFromTsv.cs

```diff
@@ -201,5 +209,43 @@ private static List<MatchedFragmentIon> ReadFragmentIonsFromString(string matche
 
             return childScanMatchedIons;
         }
+
+        private Dictionary<MatchedFragmentIon, bool> findVariantCrossingIons()
+        {
+            Dictionary<MatchedFragmentIon, bool> variantCrossingIons = new Dictionary<MatchedFragmentIon, bool>();
+
+            if (StartAndEndResiduesInProtein != null && IdentifiedSequenceVariations != null)
+            {
+                Match positionMatch = PositionParser.Match(StartAndEndResiduesInProtein);
+                Match variantMatch = VariantParser.Match(IdentifiedSequenceVariations);
+                if (positionMatch.Success && variantMatch.Success)
+                {
+                    int peptideStart = int.Parse(positionMatch.Groups[1].Value);
+                    int peptideEnd = int.Parse(positionMatch.Groups[2].Value);
+                    int variantResidueStart = int.Parse(variantMatch.Groups[1].Value);
+                    int variantResidueEnd = variantResidueStart + variantMatch.Groups[2].Value.Length - 1;
+
+                    foreach (MatchedFragmentIon ion in MatchedIons)
+                    {
+                        Match ionMatch = IonParser.Match(ion.Annotation);
+                        if (ionMatch.Success && // finds match, variant is within peptide, and ion contains variant (differs if ion is abc or xyz)
+                            (variantResidueEnd >= peptideStart && variantResidueStart <= peptideEnd) &&
+                            ((ion.NeutralTheoreticalProduct.ProductType <= ProductType.c &&
```

Comment:
this `<= ProductType.c` stuff seems weird

Reply:
I guess (ProductType <= ProductType.c) is same as (ProductType ==ProductType.a || ProductType ==ProductType.b || ProductType ==ProductType.c). Please make sure it works like what it means.

Reply:
It now checks if the ProductType is in a list of a, b, and c product types or x, y, and z product types.